### ISSUE TITLE
template_params argument

### ISSUE THREAD

**Issue by butuzov** [cweill/gotests#149]
What does [`template_params`](https://github.com/cweill/gotests/blob/develop/gotests/main.go#L60) do and how to use it? I tried to understand, but it looks that it's just redundant in terms of the `main` package. 

Or is there some hidden magic I miss?

**Comment by cweill** [comment]
It was introduced in https://github.com/cweill/gotests/pull/90, for users who want to call gotests from a separate binary.

**Comment by butuzov** [comment]
Sorry, I mean what this param does in `main.go`, as argument it does nothing. `TemplateParams` is quite clear down the road of the `output` package.

**Comment by cweill** [comment]
You're right, it looks like a bug where I forgot to pass it to the Options in #90. Would you like to send a PR to fix it?

**Comment by butuzov** [comment]
You have already assigned this to me =), but I will put this in the queue as time isn't my best friend.

**Comment by cweill** [comment]
This has been fixed! 🎉

The bug was exactly as you and @cweill identified - the `-template_params` flag was defined but never passed to the process.Options struct.

**What was fixed:**
- Added `TemplateParams` field to `process.Options`
- Updated `main.go` to pass `*templateParams` to `process.Run()`
- Updated `process.go` to parse the JSON string and use it

**Now works:**
```bash
# Using JSON string directly
$ gotests -template_params '{"key":"value"}' -all file.go

# Using JSON file (takes precedence if both are provided)
$ gotests -template_params_file params.json -all file.go
```

The fix includes proper error handling for invalid JSON and has been tested to work correctly.

Thanks for identifying this issue!
